### PR TITLE
Add dao-9 solution

### PR DESCRIPTION
**Pull request by AndreiYazadzhi**

### CODE REVIEW

**Review by MykolaFatkullin — APPROVED** [review]

File: src/main/java/com/dao/homework/dao/jdbc/CarDaoJdbcImpl.java [hunk 1]
```diff
@@ -177,9 +170,11 @@ public Manufacturer getManufacturerInstance(ResultSet resultSet) throws SQLExcep
     }
 
     public Driver getDriverInstance(ResultSet resultSet) throws SQLException {
-        Driver driver = new Driver(resultSet
-                .getObject("drivers_name", String.class),
-                resultSet.getObject("license_number", String.class));
+        String name = resultSet.getObject("drivers_name", String.class);
+        String licenseNumber = resultSet.getObject("license_number", String.class);
+        String login = resultSet.getObject("login", String.class);
+        String password = resultSet.getObject("password", String.class);
```
Comment (RomanDubovskyi) [inline src/main/java/com/dao/homework/dao/jdbc/CarDaoJdbcImpl.java]: ideally, we don't want to expose users' password when passing an object to a front end

File: src/main/java/com/dao/homework/security/AuthenticationServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private DriverService driverService;
+
+    @Override
+    public Driver login(String login, String password) {
+        Optional<Driver> driverFromDb = driverService.findByLogin(login);
+        if (driverFromDb.isPresent() && driverFromDb.get()
+                .getPassword().equals(password)
+                && driverFromDb.get().getLogin().equals(login)) {
```
Comment (RomanDubovskyi) [inline src/main/java/com/dao/homework/security/AuthenticationServiceImpl.java]:
```suggestion
                ) {
```
that will be checked in findByLogin, if login is incorrect the check will fail on the first condition

File: src/main/java/com/dao/homework/web/filters/AuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        HttpServletResponse resp = (HttpServletResponse) servletResponse;
+
+        String url = req.getServletPath();
+        if (test.contains(url)) {
+            filterChain.doFilter(req, resp);
+            return;
+        }
+
+        Long driverId = (Long) req.getSession().getAttribute(DRIVER_ID);
+        if (driverId == null || driverService.get(driverId) == null) {
```
Comment (RomanDubovskyi) [inline src/main/java/com/dao/homework/web/filters/AuthenticationFilter.java]: can the service return null?

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]
nice, almost there

File: src/main/java/com/dao/homework/web/filters/AuthenticationFilter.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AuthenticationFilter implements Filter {
+    public static final Set<String> test = new HashSet<>();
+    private static final String DRIVER_ID = "drivers_id";
+    private static final Injector injector = Injector.getInstance("com.dao.homework");
+    private final DriverService driverService = (DriverService) injector
+            .getInstance(DriverService.class);
```
Comment (RomanDubovskyi) [inline src/main/java/com/dao/homework/web/filters/AuthenticationFilter.java]: do we use it here?

File: src/main/java/com/dao/homework/web/filters/AuthenticationFilter.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AuthenticationFilter implements Filter {
+    public static final Set<String> test = new HashSet<>();
```
Comment (RomanDubovskyi) [inline src/main/java/com/dao/homework/web/filters/AuthenticationFilter.java]: what does "test" mean?

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**